src/provider/application.js

```diff
@@ -334,6 +350,84 @@ class Application extends EventEmitter {
       this.trigger('xfc.unload');
     }
   }
+
+  /**
+   * When page load is completed, get the
+   * iframe's scrolling config, and set the tabIndex
+   * on the document.body of the embedded page if
+   * necessary.
+   */
+  handleLoadEvent() {
+    this.JSONRPC.request('isScrollingEnabled', [])
+      .then(this.setFocusWhenRequired);
+  }
+
+  /**
+   * Handle the resize event and check if tabIndex is needed to be set or removed.
+   * Depending on the content is scrollable or not, we need to update the tabIndex
+   * accordingly so focus isn't getting in the document when not needed.
+   */
+  handleResizeEvent() {
+    if (this.isIframeScrollable() === true && isContentScrollable() && this.hasInteractableElement === false) {
+      // Set tabIndex="0" so focus can go into the document
+      document.body.tabIndex = 0;
+    } else if (document.body.getAttribute('tabIndex') === '0') {
+      document.body.removeAttribute('tabIndex');
+    }
+  }
+
+  /**
+   * Handle the focus event by sending a message to the frame.
+   * We don't need to set the custom outline style for focus
+   * if the document has interactable element or the content isn't scrollable.
+   */
+  handleFocusEvent() {
+    if (this.isIframeScrollable() === true && isContentScrollable() && this.hasInteractableElement === false) {
+      // Send message to the consumer/frame.js to handle `setFocus` event
+      this.JSONRPC.notification('setFocus');
+    }
+  }
+
+  /**
+   * Handle the blur event by sending a message to the frame.
+   * If the page has any interactable element, no need to set blur
+   * since the custom outline style focus wasn't set by us.
+   */
+  handleBlurEvent() {
+    if (this.hasInteractableElement) {
+      return;
+    }
+
+    // Send message to the consumer/frame.js to handle `setBlur` event
+    this.JSONRPC.notification('setBlur');
+  }
+
+  /**
+   * Return the value of iframe's scroll config stored in `isScrollingEnabled`.
+   *
+   * @returns boolean true - when the iframe is scrollable, false - when the iframe is not scrollable
+   */
+  isIframeScrollable() {
+    return this.isScrollingEnabled;
+  }
+
+  /**
+   * Sets the `tabIndex=0` on the `document.body` if required
+   * so focus can get into the embedded document.
+   *
+   * @param {*} iframeScrollingEnabled - boolean true when iframe's scrolling is true,
+   *                                             false when iframe's scrolling is false
+   */
+  setFocusWhenRequired(iframeScrollingEnabled) {
```

Comment:
This starts with `setFocus`, but it doesn't actually set focus. Maybe call it `setTabIndex...`?

Reply:
https://github.com/cerner/xfc/pull/96/commits/8f4116da61e2b7259301f47330fb6438c028d925#diff-baaccc7106fb8a5ea9ea284ba7187c11cd3235ae46e0b917305efba4161c47c6R421